### DISCUSSION TITLE
Can I parse HTML CSS using this library?

### DISCUSSION THREAD

**Discussion by Zoha131** [pointfreeco/swift-parsing#142] [Q&A]

**Comment by stephencelis** [comment]
Hi @Zoha131! `swift-parsing` is a general-purpose parsing library. It would let you build your own HTML and CSS parsers from scratch, but does not ship with HTML or CSS parsers.

I'm going to convert this to a discussion since we use issues to track bugs with the library. Thanks!

**Comment by randomeizer** [comment]
I've started implementing an XML Parser, and it's certainly possible, but is a lot of work. HTML and CSS would probably be even more, given the variety of elements available.

**Reply by RemiBardon** [reply]
Hello @randomeizer, where can I find your current implementation? Does it work already?

**Reply by RemiBardon** [reply]
(I'd like to create a KML parser-printer based on an XML parser-printer)

**Reply by randomeizer** [reply]
Hey @RemiBardon, it's still incomplete currently, unfortunately. I've been working on a couple of other projects of late. It's kinda parsing, but not into a very ergonomic or useful structure currently - it's taking the spec very literally. I would still like to make some improvements on it.

I also want to improve the error reporting before I make it public. I'd be ok letting you take a look though, if you like.

**Reply by RemiBardon** [reply]
It's not pressing, I still have a ton of other things to do before needing an XML parser-printer, but I'll keep an eye on your public implementation 👀

**Reply by randomeizer** [reply]
I'll try to remember to post it here when I do. :)

**Reply by JaapWijnen** [reply]
@RemiBardon I have an implementation of an XML parser over at https://github.com/JaapWijnen/swift-xml-parser/
Would love to hear feedback if you end up using it!

**Reply by RemiBardon** [reply]
Oh, cool 👍🏻

I will probably not have a look at it for at least 3 months, as I'm working on another project, but I'll try to remember your library 😉
